File: backend/src/main/java/com/documed/backend/attachments/Attachment.java
package com.documed.backend.attachments;

import com.documed.backend.additionalservices.AdditionalService;
import com.documed.backend.visits.Visit;
import lombok.Data;
import lombok.NonNull;

@Data
public class Attachment {
    private final int id;
    @NonNull
    private String url;
    private Visit visit;
    private AdditionalService additionalService;
}

File: frontend/src/shared/types/Attachment.ts
import { AdditionalServiceLite } from './AdditionalService';
import { VisitLite } from './Visit';

export interface Attachment extends AttachmentLite {
  visit: VisitLite | null;
  additionalService: AdditionalServiceLite | null;
}
export interface AttachmentLite {
  id: number;
  url: string;
}
